src/obstacle_params_parser.cpp

```diff
@@ -0,0 +1,172 @@
+#pragma once
+
+#include "panda_torque_mpc/obstacle_params_parser.h"
+
+#include <Eigen/Dense>
+#include <boost/shared_ptr.hpp>
+#include <fstream>
+#include <memory>
+#include <vector>
+
+#include <pinocchio/fwd.hpp>
+#include <pinocchio/multibody/model.hpp>
+
+#include <ros/ros.h>
+#include <yaml-cpp/yaml.h>
+
+namespace panda_torque_mpc {
+
+ObstacleParamsParser::ObstacleParamsParser(
+    const boost::shared_ptr<ros::NodeHandle> &pnh,
+    const boost::shared_ptr<pinocchio::GeometryModel> &collision_model)
+    : pnh_(pnh), collision_model_(collision_model){};
+
+ObstacleParamsParser::~ObstacleParamsParser(){};
+
+void ObstacleParamsParser::addCollisions() {
+  // First adding the obstacles to the geometry model
+
+  std::cout << "in the parser" << std::endl;
+
+  int obs_idx = 1;
+
+  while (pnh_->hasParam("obstacle" + std::to_string(obs_idx) + "/type")) {
+
+    std::string obstacle_name;
+    obstacle_name = "obstacle" + std::to_string(obs_idx);
```

Comment:
```suggestion
    const std::string obstacle_name = "obstacle" + std::to_string(obs_idx);
```
The name doesn't change anywhere later in the code